src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs

```diff
@@ -480,5 +481,97 @@ public class C
     public string Y;
 }");
         }
+
+        [Fact]
+        public void TestAddAttribute()
+        {
+            var code = """
+using System;
+using System.CodeAnalysis;
+
+public class C
+{
+    Type Main(Type t)
+    {
+    }
+}
+""";
+            var fixedCode = """
+using System;
+using System.CodeAnalysis;
+
+public class C
+{
+    Type Main([Example(Sample.Attribute)] Type t)
+    {
+    }
+}
+""";
+
+            var cu = SyntaxFactory.ParseCompilationUnit(code);
+            var cls = cu.Members[0];
+
+            var editor = GetEditor(cu);
+            var methodX = editor.Generator.GetMembers(cls)[0];
+
+            if (methodX is not MethodDeclarationSyntax method)
```

Comment:
Looks like this shouldn't be necessary.

Reply:
I need to cast to MethodDeclarationSyntax. Are you saying the check is unneccessary and I should just cast it in its definition?

Reply:
Yup, I think so. It can't fail, right?

Reply:
That's correct. I'll change it to a cast.